xla/backends/cpu/runtime/onednn_thunk.cc

```diff
@@ -0,0 +1,150 @@
+                                          std::move(buffers), config));
+}
+
+tsl::AsyncValueRef<Thunk::ExecuteEvent> OneDnnThunk::Execute(
+    const ExecuteParams& params) {
+  std::vector<void*> args;
+  // Prepare args
+  // 1: nargs
+  // 2: ExecutableRunOptions
+  // 3: Config
+  // 4...: Operands
+
+  // nargs
+  int64_t nargs_offset = 3;
+  int64_t num_operands = op_buffers_.arguments_shapes.size();
+  int64_t num_args = nargs_offset + num_operands;
+  args.push_back(&num_args);
+
+  // ExecutableRunOptions
+  ExecutableRunOptions run_options;
+  run_options.set_intra_op_thread_pool(params.intra_op_threadpool);
+  args.push_back(&run_options);
+
+  // Config
+  args.push_back(config_.data());
+
+  // Operands
+  std::vector<MemrefInfoHandler> memref_ptrs(num_operands);
+  for (int i = 0; i < num_operands; ++i) {
+    const auto& shape = op_buffers_.arguments_shapes[i];
+    TF_ASSIGN_OR_RETURN(se::DeviceMemoryBase arg,
+                        params.buffer_allocations->GetDeviceAddress(
+                            op_buffers_.arguments_buffers[i]));
+    memref_ptrs[i] = CreateMemrefFromShape(shape, arg.opaque());
+  }
+  for (auto& memref_ptr : memref_ptrs) {
+    args.push_back(static_cast<void*>(memref_ptr.get()));
+  }
+
+  // Prepare result
+  const auto& shape = op_buffers_.results_shapes[0];
+  TF_ASSIGN_OR_RETURN(se::DeviceMemoryBase res,
+                      params.buffer_allocations->GetDeviceAddress(
+                          op_buffers_.results_buffers[0]));
+  auto res_memref_ptr = CreateMemrefFromShape(shape, res.opaque());
+
+  // Invoke the oneDNN function based on the target.
+  if (target_ == "__onednn$convolution") {
+    __xla_cpu_runtime_OneDnnConvolution(res_memref_ptr.get(), args.data());
+  } else if (target_ == "__onednn$matmul") {
+    if (op_buffers_.is_tuple_result) {
+      // Prepare scratch
+      const auto& shape = op_buffers_.results_shapes[1];
+      TF_ASSIGN_OR_RETURN(se::DeviceMemoryBase scratch,
+                          params.buffer_allocations->GetDeviceAddress(
+                              op_buffers_.results_buffers[1]));
+      auto scratch_memref_ptr = CreateMemrefFromShape(shape, scratch.opaque());
+      __xla_cpu_runtime_OneDnnMatMul(res_memref_ptr.get(),
+                                     scratch_memref_ptr.get(), args.data());
+    } else {
+      __xla_cpu_runtime_OneDnnMatMul(res_memref_ptr.get(), nullptr,
+                                     args.data());
+    }
+  } else if (target_ == "__onednn$matmul_reorder") {
+    __xla_cpu_runtime_OneDnnMatMulReorder(res_memref_ptr.get(), args.data());
+  } else if (target_ == "__onednn$layernorm") {
+    __xla_cpu_runtime_OneDnnLayerNorm(res_memref_ptr.get(), args.data());
+  } else if (target_ == "__onednn$softmax") {
+    __xla_cpu_runtime_OneDnnSoftmax(&run_options, args[nargs_offset],
+                                    res_memref_ptr.get(), config_.data());
+  }
```

Comment:
Please add an `else` that throws an error when `target_` string is not supported.